web-common/src/features/chat/core/messages/Messages.svelte

```diff
@@ -86,108 +85,109 @@
       </div>
     </div>
   {:else}
-    {#each displayMessages as msg (msg.id)}
-      <Message
-        message={msg}
-        resultMessage={resultMessagesByParentId.get(msg.id)}
-      />
+    {#each blocks as block (block.id)}
+      {#if block.type === "text" && block.message.role === "user"}
+        <UserMessage message={block.message} />
+      {:else if block.type === "text" && block.message.role === "assistant"}
+        <AssistantMessage message={block.message} />
+      {:else if block.type === "thinking"}
+        <ThinkingBlock
+          messages={block.messages}
```

Comment:
Since this maps 1-1 with thinking block type, why not send the block as arg? Same with `ChartBlock` and `FileDiffBlock`.

Reply:
Yeah, that works too. Implemented.